lib/split/user.rb

```diff
@@ -14,12 +14,23 @@ def initialize(context, adapter=nil)
 
     def cleanup_old_experiments!
       return if @cleaned_up
-      experiment_keys(user.keys).each do |key|
-        experiment = ExperimentCatalog.find key_without_version(key)
-        if experiment.nil? || experiment.has_winner? || experiment.start_time.nil?
+      exp_to_delete = {}
+
+      user.keys.each do |key|
+        exp_name = experiment_name(key)
+
+        unless exp_to_delete.include?(exp_name)
+          experiment = ExperimentCatalog.find exp_name
+
+          if experiment.nil? || experiment.has_winner? || experiment.start_time.nil?
+            exp_to_delete[exp_name] = true
+          else
+            exp_to_delete[exp_name] = false
+          end
```

Comment:
Would it be cleaner if we change the if else statement to a 1 liner like this?
```
exp_to_delete[exp_name] = experiment.nil? || experiment.has_winner? || experiment.start_time.nil?
```

Since the condition basically determines the value of the variable?